### ISSUE TITLE
qemu: uncaught target signal 5 failed running cypress

### ISSUE THREAD

**Issue by convers39** [nektos/act#1399]
### Bug report info

```plain text
act version:            0.2.32
GOOS:                   darwin
GOARCH:                 arm64
NumCPU:                 8
Docker host:            DOCKER_HOST environment variable is unset/empty.
Sockets found:
	/var/run/docker.sock
Config files:           
	/Users/<username>/.actrc:
		-P ubuntu-latest=catthehacker/ubuntu:act-latest
		-P ubuntu-22.04=catthehacker/ubuntu:act-22.04
		-P ubuntu-20.04=catthehacker/ubuntu:act-20.04
		-P ubuntu-18.04=catthehacker/ubuntu:act-18.04
Build info:
	Go version:            go1.19.1
	Module path:           command-line-arguments
	Main version:          
	Main path:             
	Main checksum:         
	Build settings:
		-compiler:            gc
		-ldflags:             -X main.version=0.2.32
		CGO_ENABLED:          1
		CGO_CFLAGS:           
		CGO_CPPFLAGS:         
		CGO_CXXFLAGS:         
		CGO_LDFLAGS:          
		GOARCH:               arm64
		GOOS:                 darwin
Docker Engine:
	Engine version:        20.10.17
	Engine runtime:        runc
	Cgroup version:        2
	Cgroup driver:         cgroupfs
	Storage driver:        overlay2
	Registry URI:          https://index.docker.io/v1/
	OS:                    Docker Desktop
	OS type:               linux
	OS version:            
	OS arch:               aarch64
	OS kernel:             5.10.124-linuxkit
	OS CPU:                4
	OS memory:             7851 MB
	Security options:
		name=seccomp,profile=default
		name=cgroupns
```

### Command used with act

```sh
act -j e2e --container-architecture linux/amd64
```

### Describe issue

What I was trying to do is launch docker containers and run `npx cypress run` against the localhost server. But could not make it through.

Running on MacOS 12.5 with M1Pro chip.

My folder structure looks like below:

```
.
├── README.md
├── docker-compose.prod.yaml
├── docker-compose.test.yaml
├── docker-compose.yaml
├── e2e
│   ├── Dockerfile
│   ├── cypress
│   ├── cypress.config.js
│   ├── cypress.env.json
│   ├── package-lock.json
│   ├── package.json
│   └── tsconfig.json
├── backend
│   ├── Dockerfile
│   ├── package-lock.json
│   ├── package.json
│   ├── src
│   └── tsconfig.json
├── client
│   ├── Dockerfile
│   ├── README.md
│   ├── package-lock.json
│   ├── package.json
│   ├── public
│   ├── src
│   ├── tsconfig.json
│   └── yarn.lock
```

<img width="1504" alt="image" src="https://user-images.githubusercontent.com/42597395/196161677-17926291-0160-44ec-bf7d-e1c128234e0d.png">

The errors below imply issues with M1 chip, 

```
| qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| qemu: uncaught target signal 11 (Segmentation fault) - core dumped
```

Is there any workaround or am I making mistakes in running the job?

Also see the same report [here](https://github.com/cypress-io/github-action/issues/615)

NOTE: as this is a private repo I have replaced the project name and username in the logs and the workflow.

### Workflow content

```yml
name: E2E on Chrome
on: [push]
jobs:
  e2e:
    runs-on: ubuntu-20.04
    name: Cypress e2e test
    steps:
      - uses: actions/checkout@v2
      - name: Install node
        uses: actions/setup-node@v3
        with:
          node-version: 16
      - name: Install OS dependencies
        run: |
          apt search libnss3
          sudo apt-get update
          sudo apt-get install -y libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
      - name: Install Compose
        uses: KengoTODA/actions-setup-docker-compose@main
        with:
          version: "2.1.0"
      - name: Create dotenv file
        run: |
          touch .env
          echo USER_POOL_ID=${{secrets.USER_POOL_ID}} >> .env
          echo AWS_ACCESS_KEY_ID=${{secrets.AWS_ACCESS_KEY_ID}} >> .env
          echo AWS_SECRET_ACCESS_KEY=${{secrets.AWS_SECRET_ACCESS_KEY}} >> .env
          cat env
      - name: Start containers
        run: docker compose -f 'docker-compose.test.yaml' up -d --build
      - name: Run tests
        uses: cypress-io/github-action@v4
        env:
# ...
        with:
          browser: chrome
          command: npx cypress run
          working-directory: e2e
          wait-on: "http://localhost:3000, http://localhost:3001"
          spec: |
            cypress/**/*cy.{js,ts}
```

### Relevant log output

```sh
[Testing/Cypress e2e test] ⭐ Run Main Run tests
[Testing/Cypress e2e test] [DEBUG] Loading slug from git directory '/Users/<username>/Projects/<prj>/.git'
[Testing/Cypress e2e test] [DEBUG] Loading revision from git directory '/Users/<username>/Projects/<prj>/.git'
[Testing/Cypress e2e test] [DEBUG] Found revision: 632cc85421042500c4928f9bbddc52a25c9abbce
[Testing/Cypress e2e test] [DEBUG] HEAD points to '632cc85421042500c4928f9bbddc52a25c9abbce'
[Testing/Cypress e2e test] [DEBUG] using github ref: refs/heads/feature/e2e-ci
[Testing/Cypress e2e test] [DEBUG] Found revision: 632cc85421042500c4928f9bbddc52a25c9abbce
[Testing/Cypress e2e test] [DEBUG] About to run action &{Cypress.io Gleb Bahmutov GitHub Action for running Cypress end-to-end tests map[browser:{Name of the browser to use false } build:{Command to run in build step before starting tests false } cache-key:{Custom cache key false } ci-build-id:{ID associates multiple CI machines to one test run false } command:{Command that overrides cypress run false } command-prefix:{You can prefix the default test command using the command-prefix option. false } component:{Lets the action know that Cypress is running component tests and not e2e tests false false} config:{Set configuration values. Separate multiple values with a comma. The values set here override any values set in your configuration file. false } config-file:{Path to the cypress config file where configuration values are set. false } env:{Sets Cypress environment variables false } group:{Group setting for tests false } headed:{Whether or not to use headed mode false } install:{Whether or not to run install false } install-command:{Custom install command to use false } parallel:{Whether or not to load balance tests using multiple containers false } project:{Path of project to run false } quiet:{Whether or not to silence any Cypress specific output from stdout false } record:{Sends test results to Cypress Dashboard false false} runTests:{Whether or not to run tests false } spec:{Provide a specific specs to run false } start:{Command for starting local server in the background false } start-windows:{A different start command on Windows false } tag:{Tag setting for tests false } wait-on:{Local server URL to wait for false } wait-on-timeout:{Amount of time to wait for wait-on url to be available false } working-directory:{Working directory containing Cypress folder false }] map[dashboardUrl:{Cypress Dashboard URL if the run was recorded }] {node16 map[] dist/index.js  always()  always()   [] []} {green check-square}}
[Testing/Cypress e2e test] [DEBUG] Loading slug from git directory '/Users/<username>/Projects/<prj>/.git'
[Testing/Cypress e2e test] [DEBUG] Loading revision from git directory '/Users/<username>/Projects/<prj>/.git'
[Testing/Cypress e2e test] [DEBUG] Found revision: 632cc85421042500c4928f9bbddc52a25c9abbce
[Testing/Cypress e2e test] [DEBUG] HEAD points to '632cc85421042500c4928f9bbddc52a25c9abbce'
[Testing/Cypress e2e test] [DEBUG] using github ref: refs/heads/feature/e2e-ci
[Testing/Cypress e2e test] [DEBUG] Found revision: 632cc85421042500c4928f9bbddc52a25c9abbce
[Testing/Cypress e2e test] [DEBUG] Loading slug from git directory '/Users/<username>/Projects/<prj>/.git'
[Testing/Cypress e2e test] [DEBUG] Loading revision from git directory '/Users/<username>/Projects/<prj>/.git'
[Testing/Cypress e2e test] [DEBUG] Found revision: 632cc85421042500c4928f9bbddc52a25c9abbce
[Testing/Cypress e2e test] [DEBUG] HEAD points to '632cc85421042500c4928f9bbddc52a25c9abbce'
[Testing/Cypress e2e test] [DEBUG] using github ref: refs/heads/feature/e2e-ci
[Testing/Cypress e2e test] [DEBUG] Found revision: 632cc85421042500c4928f9bbddc52a25c9abbce
[Testing/Cypress e2e test] [DEBUG] type=3 actionDir=/Users/<username>/.cache/act/cypress-io-github-action@v4 actionPath= workdir=/Users/<username>/Projects/<prj> actionCacheDir=/Users/<username>/.cache/act actionName=cypress-io-github-action@v4 containerActionDir=/var/run/act/actions/cypress-io-github-action@v4
[Testing/Cypress e2e test] [DEBUG] Removing /Users/<username>/.cache/act/cypress-io-github-action@v4/.gitignore before docker cp
[Testing/Cypress e2e test] [DEBUG] /var/run/act/actions/cypress-io-github-action@v4
[Testing/Cypress e2e test]   🐳  docker cp src=/Users/<username>/.cache/act/cypress-io-github-action@v4/ dst=/var/run/act/actions/cypress-io-github-action@v4/
[Testing/Cypress e2e test] [DEBUG] Writing tarball /var/folders/33/qkcnqv0s1lg8fmqc6ndz69pm0000gn/T/act3273954999 from /Users/<username>/.cache/act/cypress-io-github-action@v4/
[Testing/Cypress e2e test] [DEBUG] Stripping prefix:/Users/<username>/.cache/act/cypress-io-github-action@v4/ src:/Users/<username>/.cache/act/cypress-io-github-action@v4/
[Testing/Cypress e2e test] [DEBUG] Extracting content from '/var/folders/33/qkcnqv0s1lg8fmqc6ndz69pm0000gn/T/act3273954999' to '/var/run/act/actions/cypress-io-github-action@v4/'
[Testing/Cypress e2e test] [DEBUG] executing remote job container: [node /var/run/act/actions/cypress-io-github-action@v4/dist/index.js]
[Testing/Cypress e2e test]   🐳  docker exec cmd=[node /var/run/act/actions/cypress-io-github-action@v4/dist/index.js] user= workdir=
[Testing/Cypress e2e test] [DEBUG] Exec command '[node /var/run/act/actions/cypress-io-github-action@v4/dist/index.js]'
[Testing/Cypress e2e test] [DEBUG] Working directory '/Users/<username>/Projects/<prj>'
[Testing/Cypress e2e test]   💬  ::debug::Resolved Keys:
[Testing/Cypress e2e test]   💬  ::debug::["npm-linux-x64-fe0d14e0f19a442c6f2191cb8c3c9414c3e52bf8a9f146df53c1573db901a0d709691392e24bf68d140ed29fa65285769aa2e4766be25dd9f8f664de75d2188d","npm-linux-x64-fe0d14e0f19a442c6f2191cb8c3c9414c3e52bf8a9f146df53c1573db901a0d709691392e24bf68d140ed29fa65285769aa2e4766be25dd9f8f664de75d2188d"]
[Testing/Cypress e2e test]   💬  ::debug::Checking zstd --version
[Testing/Cypress e2e test]   💬  ::debug::Resolved Keys:
[Testing/Cypress e2e test]   💬  ::debug::["cypress-linux-x64-fe0d14e0f19a442c6f2191cb8c3c9414c3e52bf8a9f146df53c1573db901a0d709691392e24bf68d140ed29fa65285769aa2e4766be25dd9f8f664de75d2188d","cypress-linux-x64-fe0d14e0f19a442c6f2191cb8c3c9414c3e52bf8a9f146df53c1573db901a0d709691392e24bf68d140ed29fa65285769aa2e4766be25dd9f8f664de75d2188d"]
[Testing/Cypress e2e test]   💬  ::debug::Checking zstd --version
[Testing/Cypress e2e test]   💬  ::debug::*** zstd command line interface 64-bits v1.4.4, by Yann Collet ***
[Testing/Cypress e2e test]   💬  ::debug::getCacheEntry - Attempt 1 of 2 failed with error: Cache Service Url not found, unable to restore cache.
[Testing/Cypress e2e test]   💬  ::debug::*** zstd command line interface 64-bits v1.4.4, by Yann Collet ***
[Testing/Cypress e2e test]   💬  ::debug::getCacheEntry - Attempt 1 of 2 failed with error: Cache Service Url not found, unable to restore cache.
[Testing/Cypress e2e test]   💬  ::debug::getCacheEntry - Attempt 2 of 2 failed with error: Cache Service Url not found, unable to restore cache.
[Testing/Cypress e2e test]   💬  ::debug::getCacheEntry - Attempt 2 of 2 failed with error: Cache Service Url not found, unable to restore cache.
[Testing/Cypress e2e test]   | Restoring NPM cache error: getCacheEntry failed: Cache Service Url not found, unable to restore cache.
[Testing/Cypress e2e test]   | Restoring Cypress cache error: getCacheEntry failed: Cache Service Url not found, unable to restore cache.
[Testing/Cypress e2e test]   | [command]/opt/hostedtoolcache/node/16.17.1/x64/bin/npm ci
[Testing/Cypress e2e test]   | npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
[Testing/Cypress e2e test]   | npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
[Testing/Cypress e2e test]   | npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
[Testing/Cypress e2e test]   | npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
[Testing/Cypress e2e test]   | npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
[Testing/Cypress e2e test]   | npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Testing/Cypress e2e test]   | npm WARN deprecated @aws-sdk/eventstream-marshaller@3.6.1: The package @aws-sdk/eventstream-marshaller has been renamed to @aws-sdk/eventstream-codec. Please install the renamed package.
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | added 1601 packages, and audited 1602 packages in 3m
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | 125 packages are looking for funding
[Testing/Cypress e2e test]   |   run `npm fund` for details
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | 3 moderate severity vulnerabilities
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | To address all issues (including breaking changes), run:
[Testing/Cypress e2e test]   |   npm audit fix --force
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | Run `npm audit` for details.
[Testing/Cypress e2e test]   | [command]/opt/hostedtoolcache/node/16.17.1/x64/bin/npx cypress cache list
[Testing/Cypress e2e test]   | ┌─────────┬─────────────┐
[Testing/Cypress e2e test]   | │ version │ last used   │
[Testing/Cypress e2e test]   | ├─────────┼─────────────┤
[Testing/Cypress e2e test]   | │ 10.8.0  │ a month ago │
[Testing/Cypress e2e test]   | └─────────┴─────────────┘
[Testing/Cypress e2e test]   | [command]/opt/hostedtoolcache/node/16.17.1/x64/bin/npx cypress verify
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | [STARTED] Task without title.
[Testing/Cypress e2e test]   | [FAILED] Cypress failed to start.
[Testing/Cypress e2e test]   | [FAILED] 
[Testing/Cypress e2e test]   | [FAILED] This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies
[Testing/Cypress e2e test]   | [FAILED] 
[Testing/Cypress e2e test]   | [FAILED] Please refer to the error below for more details.
[Testing/Cypress e2e test]   | [FAILED] 
[Testing/Cypress e2e test]   | [FAILED] ----------
[Testing/Cypress e2e test]   | [FAILED] 
[Testing/Cypress e2e test]   | [FAILED] qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
[Testing/Cypress e2e test]   | [FAILED] qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
[Testing/Cypress e2e test]   | [FAILED] qemu: uncaught target signal 11 (Segmentation fault) - core dumped
[Testing/Cypress e2e test]   | [FAILED] 
[Testing/Cypress e2e test]   | [FAILED] ----------
[Testing/Cypress e2e test]   | [FAILED] 
[Testing/Cypress e2e test]   | [FAILED] Platform: linux-x64 (Ubuntu - 20.04)
[Testing/Cypress e2e test]   | [FAILED] Cypress Version: 10.8.0
[Testing/Cypress e2e test]   | Cypress failed to start.
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | Please refer to the error below for more details.
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | ----------
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
[Testing/Cypress e2e test]   | qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
[Testing/Cypress e2e test]   | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | ----------
[Testing/Cypress e2e test]   | 
[Testing/Cypress e2e test]   | Platform: linux-x64 (Ubuntu - 20.04)
[Testing/Cypress e2e test]   | Cypress Version: 10.8.0
[Testing/Cypress e2e test]   ❗  ::error::The process '/opt/hostedtoolcache/node/16.17.1/x64/bin/npx' failed with exit code 1
[Testing/Cypress e2e test]   ❌  Failure - Main Run tests
[Testing/Cypress e2e test] exitcode '1': failure
```

### Additional information

I have check with `docker run --privileged --rm tonistiigi/binfmt --version` both in the workflow and my host machine([here](https://gitlab.com/qemu-project/qemu/-/issues/1079)), and they show the same result as below:

```
binfmt/a161c41 qemu/v7.0.0 go/1.18.5
```

Also I tried to uninstall and reinstall the qemu with `docker run --privileged --rm tonistiigi/binfmt --install all`, it failed with an execution error

```
cannot execute binary file: Exec format error
```

Not sure if this is relevant.

**Comment by adrienjoly** [comment]
Hi just got the same error while trying to run a Cypress action on my M1-powered MacBook Pro:

GitHub Action:

```yml
      - name: Cypress run
        uses: cypress-io/github-action@v4
```

Local Command:

```sh
$ act \
        -j "cypress-job" \
        -P large-ubuntu-runner="catthehacker/ubuntu:full-latest" \
        -s github_token="xxxxxxxxxxxxxx"
```

Resulting Logs:

```
| [STARTED] Task without title.
| [FAILED] Cypress failed to start.
| [FAILED] 
| [FAILED] This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies
| [FAILED] 
| [FAILED] Please refer to the error below for more details.
| [FAILED] 
| [FAILED] ----------
| [FAILED] 
| [FAILED] qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| [FAILED] qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| [FAILED] qemu: uncaught target signal 11 (Segmentation fault) - core dumped
| [FAILED] 
| [FAILED] ----------
| [FAILED] 
| [FAILED] Platform: linux-x64 (Ubuntu - 20.04)
| [FAILED] Cypress Version: 10.1.0
| Cypress failed to start.
| 
| This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies
| 
| Please refer to the error below for more details.
| 
| ----------
| 
| qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
| qemu: uncaught target signal 11 (Segmentation fault) - core dumped
| 
| ----------
| 
| Platform: linux-x64 (Ubuntu - 20.04)
| Cypress Version: 10.1.0
[Build, Test & Preview/Cypress against local Spree instance (using Docker)]   ❗  ::error::The process '/opt/hostedtoolcache/node/14.21.1/x64/bin/npx' failed with exit code 1
[Build, Test & Preview/Cypress against local Spree instance (using Docker)]   ❌  Failure - Main Cypress run
[Build, Test & Preview/Cypress against local Spree instance (using Docker)] exitcode '1': failure
```

**Comment by Warkanlock** [comment]
Still an issue; 

**Platform**: ARM/M1
**Workflow description**: simple workflow description using the latest ubuntu image (and correct tag for the container)

**Comment by KlentyGanesh** [comment]
Hi, I got the same error as well while trying to run a Cypress job on my M1 Macbook Air,

Command:
circleci local execute --job <My-Job-Name>

Issue Logs:

Cypress failed to start.

This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies

Please refer to the error below for more details.

----------

qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
qemu: uncaught target signal 11 (Segmentation fault) - core dumped

----------

Platform: linux-x64 (Debian - 10.6)
Cypress Version: 12.3.0
Error: 
Exited with code exit status 1

Step failed
